website/pages/docs/concepts/seal.mdx

```diff
@@ -89,75 +89,159 @@ Recovery Keys and are used for these operations.
-during the migration. For example, a migration from a cloud KMS seal to
-Shamir will require that the cloud KMS be accessible during the migration.
+during the migration. For example, a migration from a KMS seal to Shamir will
+require that the KMS key be accessible during the migration.
+
+### Migration From Shamir Seal to KMS Seal
+
+#### When **not** using Integrated Storage
 
-### Migration From Shamir to Auto Unseal
+1. Take your server cluster offline.
 
-To migrate from Shamir keys to Auto Unseal, take your server cluster offline and update
-the [seal configuration](/docs/configuration/seal) with the appropriate seal
-configuration. Bring your server back up and leave the rest of the nodes offline if
-using multi-server mode, then run the unseal process with the `-migrate` flag and bring
-the rest of the cluster online.
+2. Add the desired KMS [seal block](/docs/configuration/seal) to the
+configuration file of one of the nodes.
 
-All unseal commands must specify the `-migrate` flag. Once the required threshold of
-unseal keys are entered, unseal keys will be migrated to recovery keys.
+3. Bring this one node back up and run the unseal command by supplying the
+`-migrate` flag for each unseal key. This node will be the active node and
+completes the seal migration process. The Shamir unseal keys supplied here will
+be migrated as the recovery keys for the KMS seal.
 
-```
-$ vault operator unseal -migrate
-```
+4. Standby nodes can join the active node's cluster using the new seal blocks.
+Note that by the time standby nodes join the cluster, the migration process is
+already complete. Hence, the standby nodes will be auto-unsealed through the KMS
+key and will not require an explicit unseal command.
 
-### Migration From Auto Unseal to Shamir
+#### When using Integrated Storage
+
+Integrated Storage is implemented using the [Raft Consensus Algorithm][raft] and
+would require a quorum of nodes to be fully operational.
+
+1. Take your server cluster offline.
+
+2. Add the desired KMS [seal block](/docs/configuration/seal) to the
+configuration files of quorum number of nodes in the cluster.
+
+3. Bring the quorum number of nodes back up. On each of these nodes, run the
+unseal command by supplying the `-migrate` flag for each unseal key. The Shamir
+unseal keys supplied here will be migrated as the recovery keys for the KMS
+seal. Raft protocol will then elect a leader node and this leader node will
+complete the seal migration process. If a leadership change happens after the
+seal migration is complete, the new leader will detect the fact and does not
+repeat the seal migration.
+
+4. Any new node joining to the cluster at this point will not require manual
+running of the unseal command. The join will ensure auto-unsealing of the node
+through KMS key.
+
+### Migration From KMS Seal to Shamir Seal
 
 ~> **NOTE**: Migration to Shamir seal is not currently supported when using
```

Comment:
I think this note belongs under the top-level header - right after the first paragraph. We may want to add that KMS to same KMS is not currently supported.

Reply:
Done. Thanks for reminding about KMS to KMS of same kind. I had it in my mind to do it and it slipped through.